File: packages/lemon_js/android/src/test/kotlin/xyz/yhsj/lemon_js/LemonJsPluginTest.kt
package xyz.yhsj.lemon_js

import io.flutter.embedding.engine.plugins.FlutterPlugin
import org.mockito.Mockito
import kotlin.test.Test

internal class LemonJsPluginTest {
    @Test
    fun pluginRegistersWithoutCrashing() {
        val plugin = LemonJsPlugin()
        val binding = Mockito.mock(FlutterPlugin.FlutterPluginBinding::class.java)
        plugin.onAttachedToEngine(binding)
        plugin.onDetachedFromEngine(binding)
    }
}

File: packages/lemon_js/windows/test/lemon_js_plugin_test.cpp
#include <gtest/gtest.h>

#include "lemon_js_plugin.h"

namespace lemon_js {
namespace test {

TEST(LemonJsPlugin, CanConstruct) {
  LemonJsPlugin plugin;
  SUCCEED();
}

}  // namespace test
}  // namespace lemon_js
